vavr/src/main/java/org/jdbi/v3/vavr/VavrTupleRowMapperFactory.java

```diff
@@ -53,38 +54,55 @@
             Class<? extends Tuple> tupleClass = (Class<? extends Tuple>) erasedType;
             Array<Tuple2<Type, Integer>> tupleTypes = Array.of(tupleClass.getTypeParameters())
                     .map(tp -> resolveType(tp, type))
-                    .zipWithIndex();
+                    .zipWithIndex((t, i) -> Tuple.of(t, i + 1));
+
+            Array<Tuple3<Type, Integer, Option<String>>> withConfiguredColumnName =
+                    tupleTypes.map(t -> Tuple.of(t._1, t._2, getConfiguredColumnName(t._2, config)));
+
+            boolean anyColumnSet = withConfiguredColumnName.map(t -> t._3).exists(Option::isDefined);
+            if (anyColumnSet) {
+                Array<Optional<RowMapper<?>>> mappers = withConfiguredColumnName
+                        .map(t -> t._3.isDefined() ?
+                                getColumnMapperForDefinedColumn(t._1, t._3.get(), config) :
+                                getRowMapper(t._1, t._2, config));
+
+                boolean mappableWithConfigured = mappers.forAll(Optional::isPresent);
+                if (mappableWithConfigured) {
+                    return buildMapper(tupleClass, mappers);
+                }
```

Comment:
If any element mappers are missing, let's throw an exception that tells the user which elements we couldn't map. Could be one of a few scenarios for each missing element mapper:

1. Column name is set but column mapper not registered for the element type. Example message: "Missing column mapper for tuple element _2 of type Foo in column 'foo'"
2. Column mapper registered but column name not set. "Missing column name for tuple element _2 of type Foo" 
3. No row mapper registered for row-mapped type. "Missing row mapper for tuple element _2 of type Foo"

Otherwise the message they get is something cryptic like "No mapper registered for Tuple3<Integer,String,Foo>" with no clue why it didn't work.

Reply:
Since there is no way to find out whether the user missed a column configuration or forgot a RowMapper configuration I will opt for a more generic error message that should give enough feedback